core/trino-parser/src/main/java/io/trino/sql/tree/PatternVariable.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class PatternVariable
+        extends RowPattern
+{
+    private final Identifier name;
+
+    public PatternVariable(Identifier name)
```

Comment:
This is only used for parser tests. Use the version that includes location, so we validate that piece of information, too. (and remove this method)

Reply:
I removed this constructor and added locations in the parser test.
Please note that the locations are ignored by the assert methods anyway. I filed an issue once: https://github.com/trinodb/trino/issues/7154